modelexpress_client/python/modelexpress/source_selection_metrics.py

```
@@ -0,0 +1,194 @@
+    """
+    if not _enabled():
+        return
+    gateway = os.environ.get(ENV_PUSHGATEWAY)
+    if not gateway:
+        return
+    try:
+        from prometheus_client import REGISTRY, push_to_gateway
+
+        push_to_gateway(gateway, job=job, registry=REGISTRY)
+        logger.info("Pushed P2P metrics to %s (job=%s)", gateway, job)
+    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)